### ISSUE TITLE
TLCRecommendations.dll in nuget is delay signed.

### ISSUE THREAD

**Issue by cdhawan** [microsoft/Product-Recommendations#13]
TLCRecommendations.dll in "Microsoft.MachineLearning.TLCRecommendations:"3.8.51.1475" nuget consumed by this template project is delay signed. It won't load unless we skip the strong name validation for Microsoft public used for partially signing.

**Comment by hkusulja** [comment]
What is a future of Microsoft.MachineLearning.TLCRecommendations ? Since Microsoft delisted it from Nuget.org ? https://www.nuget.org/packages/Microsoft.MachineLearning.TLCRecommendations

Where is the source code of the dll ?

**Comment by natinimni** [comment]
Hi @cdhawan and @hkusulja , 
A signed version of TLCRecommendations.dll was just published as part of version 3.8.51.1476 of the Microsoft.MachineLearning.TLCRecommendations NuGet. 
This NuGet package was always marked as ‘unlisted’ – this means that it won’t show up in NuGet search results, but it is completely supported and available. The intention is to only make it available by directly consuming it by either the [template solution](http://aka.ms/recopcs) or by anyone who is specifically know what to look for. 
Unfortunately, The TLCRecommendations library is not an open source project.
I hope this helps.

**Comment by hkusulja** [comment]
@natinimni tx for the info.
I was wondering, why is not publicly listed this .dll, also why there is no documentation and public official info on it.
I understand this gallery and this github repo of using it, but this is just add-on value..

**Comment by natinimni** [comment]
Hi @hkusulja ,
The TLC library is not publicly listed\documented because we don’t expect it to be used as a standalone library outside the context of the gallery’s [recommendations template solution](http://aka.ms/recopcs).

**Comment by hkusulja** [comment]
Since Microsoft published this github repo., and ability to contribute, modify and create your own instance.
It should be stated that..., this is not supported, mentioned etc, officially, somewhere...., currently is not good, not clear, and wrong assumptions can be stated.

**Comment by LuisCabrer** [comment]
Thanks Hrvoje,
Nice to meet you. I am the program manager for the Recommendations Solution Template.  We really appreciate your feedback, and hope that the solution is useful to you.

As Nati has stated, we don't expect the DLL to be used standalone (at least we are not supporting that scenario). But you are right, we'll be explicit in the documentation.

Thanks, and a pleasure to meet you.

Luis Cabrera  | Cloud AI Platform | Microsoft

**Comment by hkusulja** [comment]
at least you can create a new .MD file here in github repo , and / or on public nuget, and write that this is not directly supported to be used by Microsoft, and that there is no public documentation on it.

So, additional question, if someone is using public product recommendation (this github repo), which "is supported" but it is based on not "publicly" supported .dll ...
this is confused, and I can not compare this dev model to biz model on using product recommendation or not...

**Comment by LuisCabrer** [comment]
Hi Hrovje, 

Let me clarify what support means for this solution template, as it is not an actual product per se. (The customer is not paying Microsoft for the IP and there is no actual contract between the customer and Microsoft specifically for the template usage).  When you deploy a solution template, you are essentially paying for resource utilization, not for the IP.

As such, this is what we provide:
1. We have tested the template to perform recommendations as documented. If anyone finds a bug there, we will do our best effort to fix those issues so that the template continues to be useful for our customers.
2. Once a customer modifies the code on the template, they are "on their own". That said, once again, if there is a component that does not seem to work to meet your needs (perhaps you identify something odd or unexpected), we would like to know about it.  We'll try to see if the template as shipped is impacted, and if we can repro the situation, we'll look for a fix.  This is certainly a "best effort" scenario, we are unlikely to debug customer's code.
3. The TLC dll is a black-box to you (we couldn't open source it due to non public code it uses internally). We would recommend you don't modify the way it is used today. Ideally you would interact with the Recommendations.Core.dll -- we created this abstraction for that very purpose, so you don't have to worry about how the TLC dll operates.

The short version is that this is a template that we hope you find beneficial. Many others customers have. But it is not a product per se, and it may or may not meet your business requirements.

Thanks!
Luis Cabrera

**Comment by hkusulja** [comment]
All clear and thank you.
Regarding the 3. , i understand, but it will be ok, that this blackbox have a documentation on how to use the .dll properly. you do not need to open blackbox or internal document it, or put it to open source. But if this is black box, and I am using, it, i would like to know how. Currently looking just this git repo as example usage..
